### ISSUE TITLE
Unify DDC SDK with Dart repo's SDK

### ISSUE THREAD

**Issue by jmesserly** [dart-lang/sdk#27314]
DDC compiles against the Dart SDK that is installed, but runs against an older built snapshot from our implementation. Now that we're in the same repository, we should be able to handle this in the patch_sdk script.

**Comment by jmesserly** [comment]
_From @munificent on May 4, 2016 21:14_

I'll take dart:async.

**Comment by jmesserly** [comment]
_From @vsmenon on May 5, 2016 19:17_

@jacob314 will look at dart:html once dart:async is landed

**Comment by jmesserly** [comment]
_From @munificent on May 5, 2016 22:4_

dart:async is done as of 8154650779ea579e4d599ad422e459305b405eac

**Comment by jmesserly** [comment]
_From @munificent on May 6, 2016 0:10_

Working on Uri and probably dart:convert (since Uri uses it) now.

**Comment by jmesserly** [comment]
_From @rakudrama on May 6, 2016 20:12_

Blocked by #253:
lib/collection/linked_list.dart
lib/collection/queue.dart
lib/collection/splay_map.dart

**Comment by jmesserly** [comment]
_From @munificent on May 11, 2016 0:26_

OK, Uri and dart:convert are landed, though @floitschG says there may be some more dart:convert changes coming soon after. What's left to do?

**Comment by jmesserly** [comment]
_From @munificent on May 11, 2016 23:51_

This issue was originally about *unforking* DDC's SDK but we've been using it to refer to *rolling* it. To disambiguate, I'm forking the bug. This one is now about unforking. The bug for just rolling the latest SDK is #565.

**Comment by jmesserly** [comment]
I investigated DDC summary. We can produce it with new Analyzers, but we can't build tests against it because some packages and a few tests seem to depend on dart:io (not sure if they use much of it or what's up there).

**Comment by jmesserly** [comment]
this should be way easier nowadays because we're in the same repo

**Comment by jmesserly** [comment]
https://codereview.chromium.org/2698353003/

**Comment by jmesserly** [comment]
here's one SDK fix: https://codereview.chromium.org/2718643003/

**Comment by jmesserly** [comment]
okay current list of problems.

Bob's CL here needs to be redone https://github.com/dart-lang/sdk/commit/20e402c43dfbcad502f794eecab81f311c421fb7

lots of typing problems with dart:convert, in particular "startChunkedConversion" and "bind"

need to copy over a workaround for unsound typing in ListMixin.sort (we already have this workaround in JSArray.sort)

copied over some more things from dart2js impl code (this will be in my CL)

**Comment by jmesserly** [comment]
another SDK fix https://codereview.chromium.org/2715833004/

**Comment by jmesserly** [comment]
fixed with https://github.com/dart-lang/sdk/commit/fe294f37538df7094df8b09b6bef1a53de4fc4a4 ... I'll open a new one for tracking the html libraries

**Comment by jmesserly** [comment]
new issue https://github.com/dart-lang/sdk/issues/28929

**Comment by munificent** [comment]
🎉🎉🎉🎉🎉
